### PR TITLE
Remove request IDs

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]
Looks good.

cc @JeffreyRichter

**Comment by Jinming-Hu** [comment]
@katmsft resolve conflicts after https://github.com/Azure/azure-sdk-for-cpp/pull/1931 is merged

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 - Changed return type of `DataLakeFileSystemClient::GetProperties` to `DataLakeFileSystemProperties`.
 - Changed return type of `DataLakePathClient::GetProperties` to `DataLakePathProperties`.
 - Removed `GetUserDelegationKeyResult`.
+- Removed `RequestId` from the returned results.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:19]: nit: wording consistency across blobs and datalake (see https://github.com/Azure/azure-sdk-for-cpp/pull/1931)
```suggestion
- Removed `RequestId` from return types.
```

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@
 - Renamed `GetShareFilePropertiesResult` to `ShareFileServiceProperties`.
 - `ShareClient::GetPermission` now returns `Azure::Response<std::string>`.
 - `ShareServiceClient::GetProperties` now returns `Azure::Response<Models::ShareFileServiceProperties>`.
+- Removed `RequestId` from the returned results.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:20]:
```suggestion
- Removed `RequestId` from return types.
```

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**